jest.config.js

```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  verbose: true,
+  moduleFileExtensions: ['js', 'jsx', 'ts', 'tsx'],
+  transform: {
+    '^.+\\.(ts|tsx)$': 'ts-jest'
+  },
+  globals: {
+    'ts-jest': {
+      tsConfig: {
+        ...require('./tsconfig.json').compilerOptions,
+        jsx: 'react'
+      }
+    }
+  },
+  testMatch: ['**/__tests__/**/*.spec.+(ts|tsx|js)'],
+  preset: 'ts-jest'
+};
```

Comment:
Can you try using `ts-jest`'s default setting as much as possible?

I guess you don't need to do anything apart from TypeScript's compiler option `jsx: "react"`.